gcloud-java-storage/src/main/java/com/google/gcloud/spi/DefaultStorageRpc.java

```diff
@@ -106,6 +108,15 @@ private static StorageException translate(GoogleJsonError exception) {
     return new StorageException(exception.getCode(), exception.getMessage(), retryable);
   }
 
+  private static void applyOptions(StorageObject storageObject, Map<Option, ?> options) {
```

Comment:
I wonder if we can do this change without effecting the RPC layer.

So far all Options (defined in StorageRpc) are part of the apiary request headers (e.g. see [here](https://cloud.google.com/storage/docs/json_api/v1/objects/list) and [here](https://cloud.google.com/storage/docs/json_api/v1/objects/insert)).

See my suggestion bellow on how we can achieve that but if we decide to go ahead with the current way then at least lets make this method "less" inviting. "applyOptions" sounds like a generic thing that you might want to do but this should be done only by create and write.